common/router1.cc

```diff
@@ -694,302 +736,265 @@ Router1Cfg::Router1Cfg(Context *ctx) : Settings(ctx)
+                        log("  loop: %s -> %s\n",
+                            ctx->nameOfWire(ctx->getPipSrcWire(net_info->wires.at(w).pip)),
+                            ctx->nameOfWire(w));
                 }
             }
-            log_info("final tns with respect to arc budgets: %f ns (%d nets, %d arcs)\n", tns, tns_net_count,
-                     tns_arc_count);
         }
 
-        NPNR_ASSERT(jobQueue.empty());
-        jobCache.clear();
+        bool fail = false;
 
-        for (auto &net_it : ctx->nets)
-            addNetRouteJobs(ctx, cfg, net_it.first, jobCache, jobQueue);
+        if (found_unrouted) {
+            if (ctx->debug)
+                log("check failed: found unrouted arcs\n");
+            fail = true;
+        }
 
-#ifndef NDEBUG
-        if (!jobQueue.empty()) {
-            log_info("Design strangely still contains unrouted source-sink pairs:\n");
-            while (!jobQueue.empty()) {
-                log_info("  user %d on net %s.\n", jobQueue.top().user_idx, jobQueue.top().net.c_str(ctx));
-                jobQueue.pop();
-            }
-            log_info("Checksum: 0x%08x\n", ctx->checksum());
-            ctx->check();
-            ctx->unlock();
-            return false;
+        if (found_loop) {
+            if (ctx->debug)
+                log("check failed: found loops\n");
+            fail = true;
         }
-#endif
 
-        log_info("Checksum: 0x%08x\n", ctx->checksum());
-#ifndef NDEBUG
-        ctx->check();
-#endif
-        timing_analysis(ctx, true /* slack_histogram */, true /* print_path */);
-        ctx->unlock();
-        return true;
-    } catch (log_execution_error_exception) {
-#ifndef NDEBUG
-        ctx->check();
-#endif
-        ctx->unlock();
-        return false;
+        if (found_stub) {
+            if (ctx->debug)
+                log("check failed: found stubs\n");
+            fail = true;
+        }
+
+        if (!dangling_wires.empty()) {
+            if (ctx->debug)
+                log("check failed: found dangling wires\n");
+            fail = true;
+        }
+
+        if (fail)
+            return false;
     }
+
+    return true;
 }
 
 bool Context::getActualRouteDelay(WireId src_wire, WireId dst_wire, delay_t *delay,
                                   std::unordered_map<WireId, PipId> *route, bool useEstimate)
 {
-    RipupScoreboard scores;
-    Router1Cfg cfg(this);
-    cfg.useEstimate = useEstimate;
-
-    Router router(this, cfg, scores, src_wire, dst_wire);
-
-    if (!router.routedOkay)
-        return false;
-
-    if (delay != nullptr)
-        *delay = router.visited.at(dst_wire).delay;
-
-    if (route != nullptr) {
-        WireId cursor = dst_wire;
-        while (1) {
-            PipId pip = router.visited.at(cursor).pip;
-            (*route)[cursor] = pip;
-            if (pip == PipId())
-                break;
-            cursor = getPipSrcWire(pip);
-        }
-    }
-
-    return true;
+    // FIXME
```

Comment:
Context: it appears that this is only called by the fuzzer (e.g. `ice40DelayFuzzerMain`) meaning it is not critical for this PR